### PR TITLE
Add get_mission(), get_all_missions() and get_active_mission() to client

### PR DESCRIPTION
**Pull request by WenjieWei**
added method: get all active missions, get all missions, and get the mission with specific ID.

### CODE REVIEW

File: src/interop/client.py [hunk 1]
```diff
@@ -274,6 +274,69 @@ def get_target(self, id):
         response = self._get("/api/targets/{:d}".format(id))
         target = serializers.TargetSerializer.from_json(response.json())
         return target
+	
+    def get_active_mission(self, frame):
+        """Gets active mission.
+
+	    Args:
```
Comment (anassinator) [inline src/interop/client.py]: tab

File: src/interop/client.py [hunk 2]
```diff
@@ -274,6 +274,69 @@ def get_target(self, id):
...
+        Raises:
+            Timeout: On timeout.
+            HTTPError: On request failure.
+            ConnectionError: On connection failure.
+            JSONDecodeError: On JSON decoding failure.
+            LookupError: On no active missions found.
+        """	
+        response = self._get("/api/missions")
+        for m in response.json():
+            if m["active"]:
```
Comment (anassinator) [inline src/interop/client.py]: verify you don't need == "true"

File: src/interop/client.py [hunk 3]
```diff
@@ -274,6 +274,69 @@ def get_target(self, id):
...
+             A Target ROS Message.
+
+        Raises: 
+             Timeout: On Timeout.
+             HTTPError: On request failure.
+             ConnectionError: On connection failure.
+             JSONDecodeError: On JSON decoding failure.            
+        """
+        response = self._get("/api/missions/{:d}".format(id))
+        mission = serializers.MissionDeserializer.from_json(response.json())
```
Comment (anassinator) [inline src/interop/client.py]: missing frame

File: src/interop/client.py [hunk 4]
```diff
@@ -274,6 +274,69 @@ def get_target(self, id):
...
+             m["id"]: serializers.MissionDeserializer.from_json(m, frame)
+             for m in response.json()
+        }
+        return missions
+
+    def get_mission(self, id):
+        """Returns mission with the matching ID.
+           
+        Args:
+             Target ID.
```
Comment (anassinator) [inline src/interop/client.py]: id:

File: src/interop/client.py [hunk 5]
```diff
@@ -274,6 +274,69 @@ def get_target(self, id):
...
         return target
+	
+    def get_active_mission(self, frame):
+        """Gets active mission.
+
+	    Args:
+            frame: Frame ID.
+
+        Returns
+            A Mission ROS message.
```
Comment (anassinator) [inline src/interop/client.py]: tuple of ... messages (see deserializer)

File: src/interop/client.py [hunk 6]
```diff
@@ -274,6 +274,69 @@ def get_target(self, id):
...
+        return missions
+
+    def get_mission(self, id):
+        """Returns mission with the matching ID.
+           
+        Args:
+             Target ID.
+
+        Returns:
+             A Target ROS Message.
```
Comment (anassinator) [inline src/interop/client.py]: Not a target

**Comment by WenjieWei** [comment]
Fixed the comments above.

File: src/interop/client.py [hunk 7]
```diff
@@ -278,11 +278,11 @@ def get_target(self, id):
     def get_active_mission(self, frame):
         """Gets active mission.
 
-	    Args:
+            Args:
```
Comment (anassinator) [inline src/interop/client.py]: Indentation here is still wrong

File: src/interop/client.py [hunk 8]
```diff
@@ -319,14 +319,14 @@ def get_all_missions(self, frame):
         }
         return missions
 
-    def get_mission(self, id):
+    def get_mission(self, id, frame):
         """Returns mission with the matching ID.
            
         Args:
-             Target ID.
+             Mission ID, Frame ID.
```
Comment (anassinator) [inline src/interop/client.py]: id: Mission ID.
frame: Frame ID.

File: src/interop/client.py [hunk 9]
```diff
@@ -278,11 +278,11 @@ def get_target(self, id):
     def get_active_mission(self, frame):
         """Gets active mission.
 
-	    Args:
+            Args:
             frame: Frame ID.
 
         Returns
-            A Mission ROS message.
+            flyzone, search_grid, waypoints, air_drop_pos, off_axis_targ, emergent_obj ROS message.
```
Comment (anassinator) [inline src/interop/client.py]: A tuple of (FlyZoneArray, PolygonStamped, Marker, PointStamped,
PointStamped, PointStamped) corresponding to the flyzones, search
grid, waypoints, air drop position, off axis target location, and
the emergent object location of the active mission.

File: src/interop/client.py [hunk 10]
```diff
@@ -319,14 +319,14 @@ def get_all_missions(self, frame):
...
+    def get_mission(self, id, frame):
         """Returns mission with the matching ID.
            
         Args:
-             Target ID.
+             Mission ID, Frame ID.
 
         Returns:
-             A Target ROS Message.
+             Mission corresponds to the ID.
```
Comment (anassinator) [inline src/interop/client.py]: A tuple of (FlyZoneArray, PolygonStamped, Marker, PointStamped,
PointStamped, PointStamped) corresponding to the flyzones, search
grid, waypoints, air drop position, off axis target location, and
the emergent object location of the corresponding mission.

File: src/interop/client.py [hunk 11]
```diff
@@ -274,6 +274,69 @@ def get_target(self, id):
...
+        raise LookupError("No active missions found")
+
+    def get_all_missions(self, frame):
+        """Gets all missions.
+
+        Args: 
+             frame: Frame ID.
+
+        Returns:
+             All missions.
```
Comment (anassinator) [inline src/interop/client.py]: List of tuples of (FlyZoneArray, PolygonStamped, Marker, PointStamped,
PointStamped, PointStamped) corresponding to the flyzones, search
grid, waypoints, air drop position, off axis target location, and
the emergent object location.

File: src/interop/client.py [hunk 12]
```diff
@@ -274,6 +274,69 @@ def get_target(self, id):
...
         return target
+	
+    def get_active_mission(self, frame):
+        """Gets active mission.
+
+            Args:
+            frame: Frame ID.
+
+        Returns
+            flyzone, search_grid, waypoints, air_drop_pos, off_axis_targ, emergent_obj ROS message.
```
Comment (anassinator) [inline src/interop/client.py]: A tuple of (FlyZoneArray, PolygonStamped, Marker, PointStamped,
PointStamped, PointStamped) corresponding to the flyzones, search
grid, waypoints, air drop position, off axis target location, and
the emergent object location of the active mission.

File: src/interop/client.py [hunk 13]
```diff
@@ -274,6 +274,69 @@ def get_target(self, id):
         response = self._get("/api/targets/{:d}".format(id))
         target = serializers.TargetSerializer.from_json(response.json())
         return target
+	
+    def get_active_mission(self, frame):
+        """Gets active mission.
+
+            Args:
```
Comment (anassinator) [inline src/interop/client.py]: indentation

File: src/interop/client.py [hunk 14]
```diff
@@ -274,6 +274,69 @@ def get_target(self, id):
...
+             m["id"]: serializers.MissionDeserializer.from_json(m, frame)
+             for m in response.json()
+        }
+        return missions
+
+    def get_mission(self, id, frame):
+        """Returns mission with the matching ID.
+           
+        Args:
+             Mission ID, Frame ID.
```
Comment (anassinator) [inline src/interop/client.py]: id: Mission ID.
frame: Fame ID.

File: src/interop/client.py [hunk 15]
```diff
@@ -274,6 +274,69 @@ def get_target(self, id):
...
+
+        Returns:
+             Mission corresponds to the ID.
```
Comment (anassinator) [inline src/interop/client.py]: A tuple of (FlyZoneArray, PolygonStamped, Marker, PointStamped,
PointStamped, PointStamped) corresponding to the flyzones, search
grid, waypoints, air drop position, off axis target location, and
the emergent object location of the corresponding mission.

File: src/interop/client.py [hunk 16]
```diff
@@ -274,6 +274,75 @@ def get_target(self, id):
...
+        raise LookupError("No active missions found")
+
+    def get_all_missions(self, frame):
+        """Gets all missions.
+
+        Args: 
+             frame: Frame ID.
+
+        Returns:
+             A LIST OF TUPLES OF (FlyZoneArray, PolygonStamped, Marker, PointStamped, PointStamped,
```
Comment (anassinator) [inline src/interop/client.py]: lol why they all caps, and extra space

File: src/interop/client.py [hunk 17]
```diff
@@ -274,6 +274,75 @@ def get_target(self, id):
...
+    def get_mission(self, id, frame):
+        """Returns mission with the matching ID.
+           
+        Args:
+             id: Mission ID, frame: Frame ID.
+
+        Returns:
+            A tuple of (FlyZoneArray, PolygonStamped, Marker, PointStamped, PointStamped, 
+            Pointstamped) corresponding to the flyzones, search grid, waypoints, air drop position,
+            off axis target location, and the emergent object location of the active mission.
```
Comment (anassinator) [inline src/interop/client.py]: corresponding mission

**Pull request merged**